modules/accredited_representative_portal/app/controllers/accredited_representative_portal/v0/in_progress_forms_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module AccreditedRepresentativePortal
+  module V0
+    class InProgressFormsController < ApplicationController
+      def update
+        form = find_form || build_form
+        form.update!(
+          form_data: params[:form_data],
+          metadata: params[:metadata]
+        )
+
+        render json: InProgressFormSerializer.new(form)
+      end
+
+      def show
+        form = find_form
+        render json: form&.data_and_metadata || {}
+      end
+
+      def destroy
+        form = find_form or
+          raise Common::Exceptions::RecordNotFound, params[:id]
+        form.destroy
+
+        head :no_content
+      end
+
+      private
+
+      def find_form
+        InProgressForm.form_for_user(params[:id], @current_user)
+      end
+
+      def build_form
+        build_form_for_user(params[:id], @current_user)
+      end
+
+      # NOTE: The in-progress form module can upstream this convenience that
+      # allows the caller to not know about details like legacy foreign key
+      # relations. It is totally analogous to the query convenience
+      # `form_for_user` that they expose.
+      def build_form_for_user(form_id, user)
```

Comment:
@bosawt This was the final thing we wanted to draw your attention to, but doesn't require addressing immediately. So now we believe this PR is complete and in particular takes the set of suggestions you gave throughout review.